src/cpuinfo_x86.c

```diff
@@ -25,6 +25,21 @@
 #error "Cannot compile cpuinfo_x86 on a non x86 platform."
 #endif
 
+// The following includes are necessary to provide SSE detections on pre-AVX
+// microarchitectures.
+#if defined(CPU_FEATURES_COMPILER_MSC)
+#include <windows.h>  // IsProcessorFeaturePresent
```

Comment:
I fear that by adding this header, it break __cpuindex line 73 on windows
need to investigate....

Reply:
Yep, saw that ...
I'll have a look on a Windows box.

Reply:
Mabe replace it by `processthreadsapi.h` ?
ref: https://docs.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-isprocessorfeaturepresent

Reply:
not working:
```sh
mizux@DESKTOP-SUSBN78 MINGW64 ~/work/cpu_features (fix_sse_detection_on_pre_avx_machines)
$ cmake --build build --config Debug
Microsoft (R) Build Engine version 16.6.0+5ff7b0c9e for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

  Checking Build System
  Building Custom Rule C:/Users/mizux/work/cpu_features/CMakeLists.txt
  filesystem.c
  stack_line_reader.c
  string_view.c
  Generating Code...
  utils.vcxproj -> C:\Users\mizux\work\cpu_features\build\utils.dir\Debug\utils.lib
  Building Custom Rule C:/Users/mizux/work/cpu_features/CMakeLists.txt
  cpuinfo_x86.c
C:\Users\mizux\work\cpu_features\src\cpuinfo_x86.c(73,6): warning C4391: 'Leaf __cpuidex(uint32_t,int)': incorrect return type for intrinsic function, expected 'void' [C:\Users\mizux\work\cpu_features\build\cpu_features.vcxproj]
C:\Users\mizux\work\cpu_features\src\cpuinfo_x86.c(73,6): warning C4392: 'Leaf __cpuidex(uint32_t,int)': incorrect number of arguments for intrinsic function, expected '3' arguments [C:\Users\mizux\work\cpu_features\build\cpu_features.vcxproj]
C:\Users\mizux\work\cpu_features\src\cpuinfo_x86.c(73,41): error C2169: '__cpuidex': intrinsic function, cannot be defined [C:\Users\mizux\work\cpu_features\build\cpu_features.vcxproj]
C:\Users\mizux\work\cpu_features\src\cpuinfo_x86.c(90,64): error C2168: '__cpuidex': too few actual parameters for intrinsic function [C:\Users\mizux\work\cpu_features\build\cpu_features.vcxproj]
C:\Users\mizux\work\cpu_features\src\cpuinfo_x86.c(96,32): error C2168: '__cpuidex': too few actual parameters for intrinsic function [C:\Users\mizux\work\cpu_features\build\cpu_features.vcxproj]

mizux@DESKTOP-SUSBN78 MINGW64 ~/work/cpu_features (fix_sse_detection_on_pre_avx_machines)
$ cmake --build build --config Debug
Microsoft (R) Build Engine version 16.6.0+5ff7b0c9e for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

  utils.vcxproj -> C:\Users\mizux\work\cpu_features\build\utils.dir\Debug\utils.lib
  cpuinfo_x86.c
C:\Program Files (x86)\Windows Kits\10\Include\10.0.18362.0\um\winnt.h(173,1): fatal error C1189: #error:  "No Target Architecture" [C:\Users\mizux\work\cpu_features\build\cpu_features.vcxproj]
```

Reply:
FYI it seems `__cpuidex` and `CpuIdEx` conflict (same mangle ?) try to rename it and it build
quick and dirty fix rename it CpuIdExWrap....
```cpp
Leaf CpuIdExWrap(uint32_t leaf_id, int ecx) {
  Leaf leaf;
  int data[4];
  __cpuidex(data, (int)leaf_id, ecx);
  leaf.eax = data[0];
  leaf.ebx = data[1];
  leaf.ecx = data[2];
  leaf.edx = data[3];
  return leaf;
}

uint32_t GetXCR0Eax(void) { return (uint32_t)_xgetbv(0); }

#else
#error "Unsupported compiler, x86 cpuid requires either GCC, Clang or MSVC."
#endif

static Leaf CpuId(uint32_t leaf_id) { return CpuIdExWrap(leaf_id, 0); }

static const Leaf kEmptyLeaf;

static Leaf SafeCpuIdEx(uint32_t max_cpuid_leaf, uint32_t leaf_id, int ecx) {
  if (leaf_id <= max_cpuid_leaf) {
    return CpuIdExWrap(leaf_id, ecx);
```

Reply:
Thx!